x-ui.sh

```diff
@@ -671,6 +671,163 @@ run_speedtest() {
+            else
+                iplimit_main
+            fi ;;
+        2)  
+            remove_iplimit ;;
+        3)
+            if test -f "/var/log/3xipl-banned.log"; then
+                cat /var/log/3xipl-banned.log
+            else
+                echo -e "${red}Log file not found. Please Install Fail2ban and IP Limit first.${plain}\n"
+                iplimit_main
+            fi ;;
+        *) echo "Invalid choice" ;;
+    esac
+}
+
+install_iplimit() {
+    if ! command -v fail2ban-client &>/dev/null; then
+        echo -e "${green}Fail2ban is not installed. Installing now...!${plain}\n"
+        # Check the OS and install necessary packages
+        case "${release}" in
+            ubuntu|debian)
+                sudo apt-get update && sudo apt-get install fail2ban -y ;;
+            centos)
+                sudo yum -y update && sudo yum -y install fail2ban ;;
+            fedora)
+                sudo dnf -y update && sudo dnf -y install fail2ban ;;
+            *)
+                echo -e "${red}Unsupported operating system. Please check the script and install the necessary packages manually.${plain}\n"
+                exit 1 ;;
+        esac
+        echo -e "${green}Fail2ban installed successfully!${plain}\n"
+    else
+        sudo systemctl stop fail2ban
+        echo -e "${yellow}Fail2ban is already installed.${plain}\n"
+    fi
+
+    echo -e "${green}Configuring IP Limit...${plain}\n"
+    #Check if jail.local exists
+    if ! test -f "/etc/fail2ban/jail.local"; then
+        if test -f "/etc/fail2ban/jail.conf"; then
+            sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+        else
+            echo -e "${red}File /etc/fail2ban/jail.conf not found! Probably there is something wrong with your Fail2ban installation.\nInstallation of IP Limit failed.${plain}\n"
+            exit 1
+        fi
+    fi
+
+    #Check if [3x-ipl] jail exists
+    if grep -qw '3x-ipl' /etc/fail2ban/jail.local; then
+        if test -f "/etc/fail2ban/jail.conf"; then
+            sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+        else
+            echo -e "${red}Found leftovers of previously installed IP Limit, but there's no jail.conf! Probably there is something wrong with your Fail2ban installation.\nInstallation of IP Limit failed.${plain}\n"
+            exit 1
+        fi
+    fi
+
+    #Check if log file exists
+    if ! test -f "/var/log/3xipl-banned.log"; then
+        touch /var/log/3xipl-banned.log
+    fi
+
+    #Check if service log file exists so fail2ban fail2ban won't return error
+    if ! test -f "/var/log/3xipl.log"; then
+        touch /var/log/3xipl.log
+    fi
+    
+
+    echo $'\n[3x-ipl]\nenabled=true\nfilter=3x-ipl\naction=3x-ipl\nlogpath=/var/log/3xipl.log\nmaxretry=3\nfindtime=100\nbantime=300' >> /etc/fail2ban/jail.local
+
+    echo $'[Definition]\nfailregex = [LIMIT_IP].+Email= <F-USER>.+</F-USER>.+SRC= <HOST>\nignoreregex =' > /etc/fail2ban/filter.d/3x-ipl.conf
```

Comment:
```suggestion
    cat << EOF > /etc/fail2ban/filter.d/3x-ipl.conf
[Definition]
datepattern = ^%%Y/%%m/%%d %%H:%%M:%%S
failregex   = \[LIMIT_IP\]\s*Email\s*<F-USER>\S+</F-USER>\s*\|\s*SRC\s*<ADDR>
ignoreregex =
EOF
```

Reply:
also can use this for failregex: 

```suggestion
    cat << EOF > /etc/fail2ban/filter.d/3x-ipl.conf
[Definition]
datepattern = ^%%Y/%%m/%%d %%H:%%M:%%S
failregex   = \[LIMIT_IP\] Email <F-USER>\S+</F-USER> \| SRC <ADDR>
ignoreregex =
EOF
```